src/shared/utilities/editorUtilities.ts

```diff
@@ -15,3 +16,35 @@ export function getTabSizeSetting(): number {
 export function getInlineSuggestEnabled(): boolean {
     return Settings.instance.getSection('editor').get('inlineSuggest.enabled', true)
 }
+
+export async function getOpenFilesInWindow(
+    filterPredicate?: (filePath: string) => Promise<boolean>
+): Promise<string[]> {
+    const filesOpenedInEditor: string[] = []
+
+    try {
+        const tabArrays = vscode.window.tabGroups.all
+        tabArrays.forEach(tabArray => {
+            tabArray.tabs.forEach(tab => {
+                filesOpenedInEditor.push((tab.input as any).uri.fsPath)
```

Comment:
Is there a reason not to use `vscode.window.visibleTextEditors` ?
https://github.com/aws/aws-toolkit-vscode/blob/bfd220fb09749ef345fb94c58fabb668ae9fa5b3/src/shared/utilities/vsCodeUtils.ts#L28

Reply:
yea simply `vscode.window.visibleTextEditors` won't work. It's not returning all opened files.

Reply:
// This will provide all open editors (like split view with mulitple windows)
    // with one active file from each editor.
    
    I saw this comment in the existing code.  tabs not on the top are not visible, not in the returned files.

Reply:
Does the docstring for this function mention that? That kind of context is very helpful.

Reply:
@justinmk3 not sure if it helps https://github.com/microsoft/vscode/issues/8886#issuecomment-259158438 (not a docstring in the official API)

Reply:
I can provide a screenshot as well if you want

Reply:
basically `vscode.window.visibleTextEditors` will only provide the tab which **_"users are able to see in the IDE"_**

Reply:
so if the user is not on a split view, there will be only 1 visibleTextEditor. If they are on split view, the amount of visibleTextEditor will be no more than # of split users have

Reply:
that's why we need tabGroup API here., which allow us to access all "opened" files